common/hyperscaler/rules/rules_service.go

```diff
@@ -23,5 +32,104 @@ func NewRulesServiceFromFile(rulesFilePath string) (*RulesService, error) {
 		return nil, err
 	}
 
-	return &RulesService{}, nil
+	rs := &RulesService{
+		parser:    &SimpleParser{},
+		sort:      sort,
+		unique:    unique,
+		signature: signature,
+	}
+
+	rs.Parsed = rs.parse(rulesConfig)
+	return rs, err
+}
+
+func NewRulesServiceFromString(rules string, sort, unique, signature bool) (*RulesService, error) {
+	entries := strings.Split(rules, ";")
+
+	rulesConfig := &RulesConfig{
+		Rules: entries,
+	}
+
+	rs := &RulesService{
+		parser:    &SimpleParser{},
+		sort:      sort,
+		unique:    unique,
+		signature: signature,
+	}
+
+	rs.Parsed = rs.parse(rulesConfig)
+	return rs, nil
+}
+
+func (rs *RulesService) parse(rulesConfig *RulesConfig) *ParsingResults {
+	results := NewParsingResults()
+
+	for _, entry := range rulesConfig.Rules {
+		rule, err := rs.parser.Parse(entry)
+
+		results.Apply(entry, rule, err)
+	}
+
+	if rs.sort {
+		results.Results = SortRuleEntries(results.Results)
+	}
+
+	if rs.unique {
+		results.CheckUniqueness()
+	}
+
+	if rs.signature {
+		results.CheckSignatures()
+	}
+
+	if rs.sort {
+		results.Results = SortRuleEntries(results.Results)
+	}
+
+	return results
+}
+
+func (rs *RulesService) Match(data *MatchableAttributes) map[uuid.UUID]*MatchingResult {
+	var matchingResults map[uuid.UUID]*MatchingResult = make(map[uuid.UUID]*MatchingResult)
+
+	var lastMatch *MatchingResult = nil
+	for _, result := range rs.Parsed.Results {
+		if !result.HasParsingErrors() {
+			matchingResult := &MatchingResult{
+				ParsingResultID: result.ID,
+				OriginalRule:    result.OriginalRule,
+				Rule:            result.Rule,
+			}
+
+			matchingResult.Matched = result.Rule.Matched(data)
+			if matchingResult.Matched {
+				lastMatch = matchingResult
+			}
+
+			matchingResults[result.ID] = matchingResult
+		}
+	}
+
+	if lastMatch != nil {
```

Comment:
`lastMatch` is only used to track if there was at least one `matchingResult.Matched`.
Could we just boolean flag with proper naming?

Reply:
Feel free to change however you wish.